### DISCUSSION TITLE
What's the difference between `"/post/<uuid>"` and `"/post?<uuid>"`

### DISCUSSION THREAD

**Discussion by 2-3-5-41** [rwf2/Rocket#2457] [Questions] [answered]
I'm running into a problem when trying to use a typed uri; if I use `"/post/<uuid>"`, I get a normal response with the expected outcome, but if I use `"/post?<uuid>"`, I'm met with a 404 response.

# Terminal output from `"/post?<uuid>"`
```
>> Matched: (view_post) GET /blog/post?<uuid>
>> Query string failed to match route declaration.
>> missing
>> Outcome: Forward
>> No matching routes for GET /blog/post?67e55044-10b1-426f-9247-bb680e5fe0c8 text/html.
>> No 404 catcher registered. Using Rocket default.
>> Response succeeded.
```

Is there a difference about the two typed uris that I've overlooked in the rocket documents?

**Comment by robertwayne** [comment, marked as answer]
The second one is a [Query Parameter](https://rocket.rs/v0.5-rc/guide/requests/#query-strings) - Dynamic Parameters in particular, which expect a key-value pair in the request.

To fix your issue, you would need to make your request as such: `/post?uuid=your_uuid_here`

**Reply by 2-3-5-41** [reply]
If I make the route `"/post?uuid=<uuid>"`, rust compiler spits out:
```
the trait bound `&str: FromRequest<'_>` is not satisfied
the following other types implement trait `FromRequest<'r>`:
  &'r ContentType
  &'r Host<'r>
  &'r Limits
  &'r MongoDB
  &'r Route
  &'r SecretKey
  &'r State<T>
  &'r rocket::Config
and 12 others
required for `std::option::Option<&str>` to implement `FromRequest<'_>`
```

**Reply by robertwayne** [reply]
The route should not be that, but your actual request should.

```rs
#[get("/hello/<name>")]
async fn hello_name(name: &str) -> String {
    format!("Hello, {}!\n", name)
}

#[get("/hello?<name>")]
async fn hello_query(name: &str) -> String {
    format!("Hello, {}!\n", name)
}
```

> curl 127.0.0.1:8000/hello/rob
Hello, rob!

> curl 127.0.0.1:8000/hello?name=rob
Hello, rob!

**Reply by 2-3-5-41** [reply]
Ooooh, that make more since. Thank you for the clarification
